### PR TITLE
feat: add `new` method to `Trigger`

### PR DESCRIPTION
**Pull request by tqwewe**
Adds a `new` method to `Trigger`.
This PR also implements `Default` for `Trigger` (as was suggested by clippy).

### CODE REVIEW

File: leptos_reactive/src/trigger.rs [hunk 1]
```diff
@@ -18,6 +18,47 @@ pub struct Trigger {
 }
 
 impl Trigger {
+    /// Creates a [`Trigger`](crate::Trigger), a kind of reactive primitive.
+    ///
+    /// A trigger is a data-less signal with the sole purpose
+    /// of notifying other reactive code of a change. This can be useful
+    /// for when using external data not stored in signals, for example.
+    ///
+    /// This is identical to [`create_rw_signal`].
```
Comment (gbj) [inline leptos_reactive/src/trigger.rs:27]: should read "identical to [\`create_trigger\`]" right?

Comment (tqwewe) [reply leptos_reactive/src/trigger.rs:27]: Ah sorry good catch, it should

Comment (tqwewe) [reply leptos_reactive/src/trigger.rs]: All fixed!

**Comment by gbj** [comment]
Thanks!

**Pull request merged**